src/backend/mi2/mi2.ts

```diff
@@ -393,6 +393,7 @@ export class MI2 extends EventEmitter implements IBackend {
             // program is in paused state
             try {
                 await this.sendCommand('target-disconnect');
+                await new Promise(f => setTimeout(f, 250));
```

Comment:
I don't see how or why this works.  I ask because target-disconnect is failing, causing an exception and going to line 400 (catch block). So, the timeout has no effect -- didn't even get executed.

See [comment](https://github.com/Marus/cortex-debug/issues/629#issuecomment-1128152613)

Reply:
Yeah, that is a good idea looking back. But now, it is too late.

This is so frustrating. the end of the session is rife with landmines. Things fail at every possible point. And for not explainable reason. I can accept your solution as is (with some analysis) and update #629

Now, there are two+ timeouts in this code path. My worry is that they conflict.

Reply:
It would have been nice if the issues were due to the same origin.

I totally can feel that frustration. I hope there are still hints coming up on that topic.

Nevertheless thanks for analysing the PR.

Reply:
Just to be sure, which gdb-server are you using? I cannot get the timeout to do anything for me with JLink.